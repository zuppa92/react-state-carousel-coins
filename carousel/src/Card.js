import React from "react";
import "./Card.css";

function Card({ caption, src, currNum, totalNum }) {
  return (
    <div className="Card">
      <img src={src} alt={caption} />
      <p>{caption}</p>
      <p>
        Image {currNum} of {totalNum}
      </p>
    </div>
  );
}

export default Card;
