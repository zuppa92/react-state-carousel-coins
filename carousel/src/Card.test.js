import { render } from "@testing-library/react";
import Card from "./Card";

it("renders without crashing", function() {
  render(<Card caption="test caption" src="test.jpg" currNum={1} totalNum={3} />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<Card caption="test caption" src="test.jpg" currNum={1} totalNum={3} />);
  expect(asFragment()).toMatchSnapshot();
});
