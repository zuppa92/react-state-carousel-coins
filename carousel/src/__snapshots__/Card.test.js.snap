// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <img
      alt="test caption"
      src="test.jpg"
    />
    <p>
      test caption
    </p>
    <p>
      Image 1 of 3
    </p>
  </div>
</DocumentFragment>
`;
